plugins/services/src/js/reducers/serviceForm/FormReducers/SingleContainerPortMappingsReducer.js

```diff
@@ -0,0 +1,76 @@
+import { ADD_ITEM, REMOVE_ITEM, SET } from "#SRC/js/constants/TransactionTypes";
+import { PROTOCOLS } from "../../../constants/PortDefinitionConstants";
+
+const FIELDS = [
+  "automaticPort",
+  "containerPort",
+  "hostPort",
+  "labels",
+  "loadBalanced",
+  "portMapping",
+  "name",
+  "servicePort",
+  "vip"
+];
+
+/**
+ * @param {Object[]} state
+ * @param {Object} action
+ * @param {(ADD_ITEM|REMOVE_ITEM|SET)} action.type - action to perform
+ * @param {String[]} action.path - location of value
+ * @param {*} action.value - to perform action with
+ * @return {Object[]} new portDefinitions with action performed on it
+ */
+function SingleContainerPortMappingsReducer(state = [], action) {
+  const { type, path = [], value } = action;
+  const [base, index, field, subfield] = path;
+  const joinedPath = path.join(".");
+
+  if (base === "portDefinitions") {
```

Comment:
Not blocking as you "only" separated the reducers, but we need to reduce the complexity of these reducers as it's really hard to read and understand them.